### ISSUE TITLE
InterfaceType decorator with name property defined generates schema error

### ISSUE THREAD

**Issue by icereval** [nestjs/graphql#1030]
## I'm submitting a...

<pre><code>
[ ] Regression 
[x] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior

`@InterfaceType` decorator with `name` property defined generates schema error when using `@ResolveField` within an interface.

```
Error: "IRecipe" defined in resolvers, but not in schema
```

## Expected behavior

When the `name` property is defined on `@InterfaceType` the `@ResolveField` decorator should find the decorator name, rather than the interface class name.

## Minimal reproduction of the problem with instructions

https://github.com/icereval/nestjs-graphql-interface-resolver-issue

## What is the motivation / use case for changing the behavior?

To follow existing functionality provided by `@ObjectType`.

Related change: https://github.com/nestjs/graphql/commit/5949d0d7e0c4594bb9c07162baa56671f7550cf5

## Environment

<pre><code>
Nest version: 7.3.2
Nest GraphQL version: 7.5.5

 
For Tooling issues:
- Node version: 14  
- Platform:  Mac 

Others:

</code></pre>

**Comment by kamilmysliwiec** [comment]
Can you please test with `@nestjs/graphql@next`?

**Comment by icereval** [comment]
Upgraded to `next` w/ https://github.com/icereval/nestjs-graphql-interface-resolver-issue/commit/511e4d0c07cddac920dae1c3aafb1751f6b7318a

Can confirm the issue still occurs.

```
(node:61028) UnhandledPromiseRejectionWarning: Error: "IRecipe" defined in resolvers, but not in schema
```

**Comment by kamilmysliwiec** [comment]
Fixed in 7.5.6
